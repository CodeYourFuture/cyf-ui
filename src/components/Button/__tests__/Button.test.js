import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Button } from "..";

describe("Button", () => {
  test("matches snapshots for primary", () => {
    const mockOnClick = jest.fn(() => null);

    const { container } = render(
      <Button onClick={mockOnClick} variant="primary">
        Click Me
      </Button>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test("onClick is called upon user click", () => {
    const mockOnClick = jest.fn(() => null);

    const { getByText } = render(
      <Button label="Button" onClick={mockOnClick} variant="secondary">
        Click Me
      </Button>
    );

    const button = getByText("Click Me");
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalled();
  });
});
