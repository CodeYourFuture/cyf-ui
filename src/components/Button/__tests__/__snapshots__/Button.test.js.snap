// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshots for primary 1`] = `
<button
  class="sc-bdVaJa euJmou"
>
  Click Me
</button>
`;
